opentech/static_src/src/javascript/apply/edit-comment.js

```diff
@@ -0,0 +1,131 @@
+    const editBlock = '.js-edit-block';
+    const lastEdited = '.js-last-edited';
+    const editButton = '.js-edit-comment';
+    const feedContent = '.js-feed-content';
+    const commentError = '.js-comment-error';
+    const cancelEditButton = '.js-cancel-edit';
+    const submitEditButton = '.js-submit-edit';
+
+    // handle edit
+    $(editButton).click(function (e) {
+        e.preventDefault();
+
+        closeAllEditors();
+
+        const editBlockWrapper = $(this).closest(feedContent).find(editBlock);
+        const commentWrapper = $(this).closest(feedContent).find(comment);
+        const commentContents = $(commentWrapper).attr('data-comment');
+
+        // hide the edit link and original comment
+        $(this).parent().hide();
+        $(commentWrapper).hide();
+
+        const markup = `
+            <div class="js-pagedown form">
+                <div id="wmd-button-bar-edit-comment" class="wmd-button-bar"></div>
+                <textarea id="wmd-input-edit-comment" class="wmd-input" rows="10">${commentContents}</textarea>
+                <div id="wmd-preview-edit-comment" class="wmd-preview"></div>
+                <div class="wrapper--outer-space-medium">
+                    <button class="button button--primary js-submit-edit" type="submit">Update</button>
+                    <button class="button button--white js-cancel-edit">Cancel</button>
+                </div>
+            </div>
+        `;
+
+        // add the comment to the editor
+        $(editBlockWrapper).append(markup);
+
+        // run the editor
+        initEditor();
+    });
+
+    // handle cancel
+    $(document).on('click', cancelEditButton, function () {
+        showComment(this);
+        showEditButton(this);
+        hidePageDownEditor(this);
+        if ($(commentError).length) {
+            hideError();
+        }
+    });
+
+    // handle submit
+    $(document).on('click', submitEditButton, function () {
+        const commentContainer = $(this).closest(editBlock).siblings(comment);
+        const editedComment = $(this).closest(pageDown).find('.wmd-preview').html();
+        const commentMD = $(this).closest(editBlock).find('textarea').val();
+        const editUrl = $(commentContainer).attr('data-edit-url');
+
+        fetch(editUrl, {
+            method: 'POST',
+            headers: {
+                'Content-Type': 'application/json',
+                'X-CSRFToken': $.cookie('csrftoken')
+            },
+            body: JSON.stringify({
+                message: editedComment
+            })
+        }).then(response => {
+            if (response.status === 404) {
+                $(this).closest(editBlock).append('<p class="wrapper--error js-comment-error"><b>Update unsuccessful</b>. This comment has been edited elsewhere - please refresh to get the latest updates.</p>');
+                $(this).attr('disabled', true);
+                return;
+            }
+            return response.json();
```

Comment:
Do we need to handle 500 or catch generally? network issues etc?

Reply:
Definitely. I've caught all other non-404 errors.